frappe/database/mariadb/setup_db.py

```diff
@@ -1,148 +1,167 @@
-def check_if_ready_for_barracuda():
-	mariadb_variables = frappe._dict(frappe.db.sql("""show variables"""))
-	mariadb_minor_version = int(mariadb_variables.get('version').split('-')[0].split('.')[1])
-	if mariadb_minor_version < 3:
-		check_database(mariadb_variables, {
-			"innodb_file_format": "Barracuda",
-			"innodb_file_per_table": "ON",
-			"innodb_large_prefix": "ON"
-		})
-	check_database(mariadb_variables, {
-		"character_set_server": "utf8mb4",
-		"collation_server": "utf8mb4_unicode_ci"
-	})
-
-def check_database(mariadb_variables, variables_dict):
-	mariadb_minor_version = int(mariadb_variables.get('version').split('-')[0].split('.')[1])
-	for key, value in variables_dict.items():
-		if mariadb_variables.get(key) != value:
-			site = frappe.local.site
-			msg = ("Creation of your site - {x} failed because MariaDB is not properly {sep}"
-				   "configured to use the Barracuda storage engine. {sep}"
-				   "Please add the settings below to MariaDB's my.cnf, restart MariaDB then {sep}"
-				   "run `bench new-site {x}` again.{sep2}"
-				   "").format(x=site, sep2="\n"*2, sep="\n")
-
-			if mariadb_minor_version < 3:
-				print_db_config(msg, expected_config_for_barracuda_2)
-			else:
-				print_db_config(msg, expected_config_for_barracuda_3)
-			raise frappe.exceptions.ImproperDBConfigurationError(
-				reason="MariaDB default file format is not Barracuda"
-			)
+    if verbose:
+        print("Starting database import...")
+    db_name = frappe.conf.db_name
+    if not source_sql:
+        source_sql = os.path.join(os.path.dirname(
+            __file__), 'framework_mariadb.sql')
+    DbManager(frappe.local.db).restore_database(
+        db_name, source_sql, db_name, frappe.conf.db_password)
+    if verbose:
+        print("Imported from database %s" % source_sql)
+
+
+def check_database_settings():
+    versions = get_mariadb_versions()
+    if versions['major'] <= '10.2':
+        expected_variables = expected_settings_10_2_earlier
+    else:
+        expected_variables = expected_settings_10_3_later
+
+    mariadb_variables = frappe._dict(frappe.db.sql("""show variables"""))
+    # Check each expected value vs. actuals:
+    result = True
+    for key, expected_value in expected_variables.items():
+        if mariadb_variables.get(key) != expected_value:
+            print("For key %s. Expected value %s, found value %s" %
+                  (key, expected_value, mariadb_variables.get(key)))
+            result = False
+    if not result:
+        site = frappe.local.site
+        msg = ("Creation of your site - {x} failed because MariaDB is not properly {sep}"
+               "configured.  If using version 10.2.x or earlier, make sure you use the {sep}"
+               "the Barracuda storage engine. {sep}{sep}"
+               "Please verify the settings below in MariaDB's my.cnf.  Restart MariaDB.  And {sep}"
+               "then run `bench new-site {x}` again.{sep2}"
+               "").format(x=site, sep2="\n"*2, sep="\n")
+
```

Comment:
is there something missing over here?

Reply:
Oh, you're missing `print_db_config()` in here.

Reply:
Ah, you're right!  What's the best way for me to fix that?  Do I need close and re-submit a brand new pull?

Reply:
No, just push more commits to this branch. It should show up here.

Reply:
Excellent, thanks for that info.  :)  I've added the missing line to call print_db_config().